compiler/rustc_expand/src/base.rs

```diff
@@ -1264,17 +1287,23 @@ pub fn expr_to_spanned_string<'a>(
     cx: &'a mut ExtCtxt<'_>,
     expr: P<ast::Expr>,
     err_msg: &'static str,
-) -> Result<
-    (Symbol, ast::StrStyle, Span),
-    Result<(Diag<'a>, bool /* has_suggestions */), ErrorGuaranteed>,
-> {
+) -> ExpandResult<ExprToSpannedStringResult<'a>, ()> {
+    if let ast::ExprKind::MacCall(m) = &expr.kind
+        && !cx.force_mode
+        && cx.resolver.cfg_accessible(cx.current_expansion.id, &m.path).is_err()
+    {
+        return ExpandResult::Retry(());
+    }
+
     // Perform eager expansion on the expression.
     // We want to be able to handle e.g., `concat!("foo", "bar")`.
     let expr = cx.expander().fully_expand_fragment(AstFragment::Expr(expr)).make_expr();
```

Comment:
Ideally, the whole `fully_expand_fragment` would return Ready/Retry, but that's probably a much larger change.
The logic above is a good enough approximation.